### PR TITLE
Fleshes out the architectural components

### PR DESCRIPTION
**Pull request by mhausenblas**
CC: @mrunalp

### CODE REVIEW

File: index.md [hunk 1]
```diff
@@ -23,6 +27,52 @@ https://github.com/kubernetes-incubator/cri-o
...
+The [containers/image](https://github.com/containers/image) lib is used for pulling images from registries.
+Currently, it supports Docker [schema 2/version 1](https://docs.docker.com/registry/spec/manifest-v2-1/)
+as well as [schema 2/version 2](https://docs.docker.com/registry/spec/manifest-v2-2/).
+It also passes all Docker and Kubernetes tests.
+
+### Networking
+The Container Network Interface [CNI](https://github.com/containernetworking/cni) is used for setting up networking for the pods.
+Various CNI plugins such as Flannel, Weave and OpenShift-SDN have been tested with CRI-O and are working as expected.
+
+### Monitoring
```
Comment (runcom) [inline index.md]: I would call this something like "handling container lifecycle and utils" instead of just monitoring, @mrunalp wdyt?

Comment (mhausenblas) [inline index.md]: Fair enough. FYI: this is exactly the wording we're using in the blog post ;)

File: index.md [hunk 2]
```diff
@@ -23,6 +27,52 @@ https://github.com/kubernetes-incubator/cri-o
...
+
+[conmon](https://github.com/kubernetes-incubator/cri-o/tree/master/conmon) is a utility within CRI-O that is used to
+monitor the containers, handle logging from the container process, serve attach clients and detects Out Of Memory (OOM)
+situations.
+
+### Security
+
+Container security separation policies are provided by a series of tools including SELinux, Capabilities, seccomp,
+and other security separation policies as specified in the OCI Specification.
```
Comment (aweiteka) [inline index.md]: So... basically Linux. Should we call this out? A "Containers are Linux" kind of thing. Or... "Workload separation" == "leveraging existing Linux features".

**Comment by runcom** [comment]
as talked with @mhausenblas on IRC, let's get this is and address the comments in a follow up

**Pull request merged**